common/namer/namer.go

```diff
@@ -55,3 +61,22 @@ func (n Namer) DisplayName(parts ...pulumi.StringInput) pulumi.StringInput {
 		return n.ResourceName(strArgs...)
 	}).(pulumi.StringOutput)
 }
+
+// HashedName return the md5 32-chars hex pulumi.StringInput of `parts` prefixed
+// with the namer prefix
+func (n Namer) HashedName(parts ...pulumi.StringInput) pulumi.StringInput {
+	return n.DisplayName(parts...).ToStringOutput().ApplyT(func(name string) string {
+		hmd5 := md5.Sum([]byte(name))
+		return hex.EncodeToString(hmd5[:])
+	}).(pulumi.StringOutput)
+}
+
+// DisplayHashedName return the first 15 chars of the display name plus the first 16 chars
+// of the md5 32-chars hex pulumi.StringInput of display name
+func (n Namer) DisplayHashedName(parts ...pulumi.StringInput) pulumi.StringInput {
+	return pulumi.All(n.DisplayName(parts...), n.HashedName(parts...)).ApplyT(func(args []interface{}) string {
+		displayName := args[0].(string)
+		hashedName := args[1].(string)
```

Comment:
If `len(displayName) <= 32`, don't you want to keep the full `displayName`?

Reply:
Right !